### DISCUSSION TITLE
Add support for validating addresses in the context of an order

### DISCUSSION THREAD

**Discussion by lukeholder** [craftcms/commerce#2445] [Ideas]
Access to the order model when validating addresses, as well as access to the scenario (adding new, updating etc) should make it easier for developers to add validation rules to addresses that works in different contexts.

**Comment by Mosnar** [comment]
@lukeholder I was about to open an issue for a scenario, but I wanted to see if this issue would cover it. Here's the scenario:

We added address validation via `EVENT_DEFINE_RULES` on the Address model; however, this broke adding items to the cart for customers who had an address set on the order that doesn't pass validation. I'm not sure the best work-around, but as quick-fix, I added a check to see if shippingAddress or billingAddress are present in the body params and only attaching the rules if so.

**Comment by pdaleramirez** [comment]
@Mosnar You can do this as a workaround

```
        Event::on(
            Address::class,
            Model::EVENT_DEFINE_RULES,
            function(DefineRulesEvent $event) {
                if (Craft::$app->requestedAction->actionMethod !== 'actionUpdateCart') {
                    $rules = $event->rules;
                    $rules[] = [['firstName', 'lastName'], 'required'];
                    $event->rules = $rules;
                }
            }
        );
```
